IllegalArgumentException: ReadTimeout must not be null

При отсутсвии параметра spring.ai.gigachat.internal.read-timeout приложение не запускается.

```
Caused by: java.lang.IllegalArgumentException: ReadTimeout must not be null
at org.springframework.util.Assert.notNull(Assert.java:172)
at org.springframework.http.client.JdkClientHttpRequestFactory.setReadTimeout(JdkClientHttpRequestFactory.java:98)
at chat.giga.springai.api.chat.GigaChatApi.<init>(GigaChatApi.java:86)
at chat.giga.springai.autoconfigure.GigaChatAutoConfiguration.gigaChatApi(GigaChatAutoConfiguration.java:106)
```

Окружение:
Spring Boot 3.3.0
spring-ai-gigachat 1.1.0-M1.001
spring-ai 1.0.0-M2

--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@
 <dependency>
     <groupId>chat.giga</groupId>
     <artifactId>spring-ai-starter-model-gigachat</artifactId>
-    <version>1.0.5</version>
+    <version>1.0.6</version>
 </dependency>
 ```
 

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>chat.giga</groupId>
     <artifactId>spring-ai-gigachat-parent</artifactId>
-    <version>1.0.5</version>
+    <version>1.0.6</version>
     <packaging>pom</packaging>
 
     <name>Spring AI model - GigaChat - Parent</name>

--- a/pom.xml
+++ b/pom.xml
@@ -60,7 +60,7 @@
         <allure-bom.version>2.29.1</allure-bom.version>
         <aspectj.version>1.9.24</aspectj.version>
         <spring-boot.version>3.5.5</spring-boot.version>
-        <spring-ai.version>1.0.2</spring-ai.version>
+        <spring-ai.version>1.0.3</spring-ai.version>
         <ayza.version>10.0.0</ayza.version>
         <lombok.version>1.18.38</lombok.version>
         <wiremock-spring-boot.version>3.10.6</wiremock-spring-boot.version>

--- a/spring-ai-autoconfigure-model-gigachat/pom.xml
+++ b/spring-ai-autoconfigure-model-gigachat/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>chat.giga</groupId>
         <artifactId>spring-ai-gigachat-parent</artifactId>
-        <version>1.0.5</version>
+        <version>1.0.6</version>
     </parent>
 
     <artifactId>spring-ai-autoconfigure-model-gigachat</artifactId>

--- a/spring-ai-gigachat-example/pom.xml
+++ b/spring-ai-gigachat-example/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>chat.giga</groupId>
     <artifactId>spring-ai-gigachat-example</artifactId>
-    <version>1.0.5</version>
+    <version>1.0.6</version>
 
     <parent>
         <groupId>org.springframework.boot</groupId>

--- a/spring-ai-gigachat-example/pom.xml
+++ b/spring-ai-gigachat-example/pom.xml
@@ -17,7 +17,7 @@
 
     <properties>
         <java.version>17</java.version>
-        <spring-ai.version>1.0.2</spring-ai.version>
+        <spring-ai.version>1.0.3</spring-ai.version>
     </properties>
 
     <dependencyManagement>

--- a/spring-ai-gigachat/pom.xml
+++ b/spring-ai-gigachat/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>chat.giga</groupId>
         <artifactId>spring-ai-gigachat-parent</artifactId>
-        <version>1.0.5</version>
+        <version>1.0.6</version>
     </parent>
 
     <artifactId>spring-ai-gigachat</artifactId>

--- a/spring-ai-gigachat/src/main/java/chat/giga/springai/api/auth/bearer/GigaChatBearerAuthApi.java
+++ b/spring-ai-gigachat/src/main/java/chat/giga/springai/api/auth/bearer/GigaChatBearerAuthApi.java
@@ -3,6 +3,7 @@ package chat.giga.springai.api.auth.bearer;
 import static chat.giga.springai.api.chat.GigaChatApi.USER_AGENT_SPRING_AI_GIGACHAT;
 
 import chat.giga.springai.api.GigaChatApiProperties;
+import chat.giga.springai.api.GigaChatInternalProperties;
 import chat.giga.springai.api.HttpClientUtils;
 import chat.giga.springai.api.auth.GigaChatApiScope;
 import com.fasterxml.jackson.annotation.JsonProperty;
@@ -45,9 +46,12 @@ public class GigaChatBearerAuthApi {
         boolean isUnsafeSsl = apiProperties.isUnsafeSsl();
         SSLFactory sslFactory = HttpClientUtils.buildSslFactory(kmf, tmf, isUnsafeSsl);
         String authUrl = apiProperties.getAuthUrl();
+        GigaChatInternalProperties internalProps = apiProperties.getInternal();
         var clientHttpRequestFactory = new JdkClientHttpRequestFactory(HttpClientUtils.buildHttpClient(
-                sslFactory, apiProperties.getInternal().getConnectTimeout()));
-        clientHttpRequestFactory.setReadTimeout(apiProperties.getInternal().getReadTimeout());
+                sslFactory, internalProps.getConnectTimeout()));
+        if (internalProps.getReadTimeout() != null) {
+            clientHttpRequestFactory.setReadTimeout(internalProps.getReadTimeout());
+        }
         this.restClient = builder.baseUrl(authUrl)
                 .requestFactory(clientHttpRequestFactory)
                 .build();

--- a/spring-ai-gigachat/src/main/java/chat/giga/springai/api/auth/bearer/GigaChatBearerAuthApi.java
+++ b/spring-ai-gigachat/src/main/java/chat/giga/springai/api/auth/bearer/GigaChatBearerAuthApi.java
@@ -47,8 +47,8 @@ public class GigaChatBearerAuthApi {
         SSLFactory sslFactory = HttpClientUtils.buildSslFactory(kmf, tmf, isUnsafeSsl);
         String authUrl = apiProperties.getAuthUrl();
         GigaChatInternalProperties internalProps = apiProperties.getInternal();
-        var clientHttpRequestFactory = new JdkClientHttpRequestFactory(HttpClientUtils.buildHttpClient(
-                sslFactory, internalProps.getConnectTimeout()));
+        var clientHttpRequestFactory = new JdkClientHttpRequestFactory(
+                HttpClientUtils.buildHttpClient(sslFactory, internalProps.getConnectTimeout()));
         if (internalProps.getReadTimeout() != null) {
             clientHttpRequestFactory.setReadTimeout(internalProps.getReadTimeout());
         }

--- a/spring-ai-gigachat/src/main/java/chat/giga/springai/api/chat/GigaChatApi.java
+++ b/spring-ai-gigachat/src/main/java/chat/giga/springai/api/chat/GigaChatApi.java
@@ -83,7 +83,9 @@ public class GigaChatApi {
 
         var clientHttpRequestFactory = new JdkClientHttpRequestFactory(buildHttpClient(
                 buildSslFactory(kmf, tmf, properties.isUnsafeSsl()), internalProps.getConnectTimeout()));
-        clientHttpRequestFactory.setReadTimeout(internalProps.getReadTimeout());
+        if (internalProps.getReadTimeout() != null) {
+            clientHttpRequestFactory.setReadTimeout(internalProps.getReadTimeout());
+        }
         this.restClient = restClientBuilder
                 .clone()
                 .requestFactory(clientHttpRequestFactory)

--- a/spring-ai-starter-model-gigachat/pom.xml
+++ b/spring-ai-starter-model-gigachat/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>chat.giga</groupId>
         <artifactId>spring-ai-gigachat-parent</artifactId>
-        <version>1.0.5</version>
+        <version>1.0.6</version>
     </parent>
 
     <artifactId>spring-ai-starter-model-gigachat</artifactId>

